blog/resources/views/changepassword/change.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+use App\Auth\Form\ChangePasswordForm;
+use Yiisoft\FormModel\Field;
+use Yiisoft\Html\Html;
+use Yiisoft\Html\Tag\Form;
+use Yiisoft\Router\UrlGeneratorInterface;
+use Yiisoft\Translator\TranslatorInterface;
+use Yiisoft\View\WebView;
+
+/**
+ * @var WebView               $this
+ * @var TranslatorInterface   $translator
+ * @var UrlGeneratorInterface $urlGenerator
+ * @var string                $csrf
+ * @var ChangePasswordForm    $formModel
+ */
+$this->setTitle($translator->translate('layout.change-password'));
+?>
+
+<div class="container py-5 h-100">
+    <div class="row d-flex justify-content-center align-items-center h-100">
+        <div class="col-12 col-md-8 col-lg-6 col-xl-5">
+            <div class="card border border-dark shadow-2-strong rounded-3">
+                <div class="card-header bg-dark text-white">
+                    <h1 class="fw-normal h3 text-center"><?= Html::encode($this->getTitle()) ?></h1>
+                </div>
+                <div class="card-body p-5 text-center">
+                    <?= Form::tag()
+                        // note: the change function actually appears in the ChangeController
+                        ->post($urlGenerator->generate('auth/change'))
+                        ->csrf($csrf)
+                        ->id('changeForm')
+                        ->open() ?>
+
+                    <?= Field::text($formModel, 'login')->addInputAttributes(['value'=> $login ?? '', 'readonly'=>'readonly']) ?>
```

Comment:
Do you want add ability to change password for any user? Not only current?

Reply:
Yes. I will need to add a permission to the readonly => readonly  e.g.

!$this->currentUser->can('changePasswordForAnyUser',[]) ?  'readonly'=>'readonly' :  '';

This will make sure that only administrators with the canChangePasswordForAnyUser permission can insert a username in the login textbox in place of the {login} in order to change the password.

...And remove the

````
if ($this->currentUser->can('viewInv',[])) {
````  

since all authenticated users with a lesser permission of viewInv should be able to change their password.

changepassword.php
````
<?= $canChangePasswordForAnyUser ? Field::text($formModel, 'login')->addInputAttributes(['value'=> $login ?? '']) 
                                                     : Field::text($formModel, 'login')->addInputAttributes(['value'=> $login ?? '', 'readonly'=>'readonly']); ?>
````